File: multi/infrastructure/src/main/java/com/example/infrastructure/screening/ScreeningJpaRepository.java
package com.example.infrastructure.screening;

import com.example.domain.screening.Screening;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.data.jpa.repository.Query;
import org.springframework.stereotype.Repository;

import java.util.List;

@Repository
public interface ScreeningJpaRepository extends JpaRepository<Screening,Long> {

    @Query("select s from Screening s join fetch s.movie join fetch s.theater order by s.movie.releaseDate, s.screeningDate, s.startTime ")
    List<Screening> findAll();

    Screening findByMovieId(Long movieId);
}

File: movie/infra/src/main/kotlin/com/example/movie/persistence/screening/repository/ScreeningJpaRepository.kt
package com.example.movie.persistence.screening.repository

import com.example.movie.persistence.screening.model.ScreeningEntity
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository

@Repository
interface ScreeningJpaRepository : JpaRepository<ScreeningEntity, Long> {
}